packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -474,22 +474,27 @@ class Padding extends SingleChildRenderObjectWidget {
   }
 }
 
-/// Aligns its child box within itself.
+/// Aligns its child within itself and optionally sizes itself based on the
+/// child's size.
 ///
 /// For example, to align a box at the bottom right, you would pass this box a
 /// tight constraint that is bigger than the child's natural size,
 /// with an alignment of [FractionalOffset.bottomRight].
 ///
-/// By default, sizes to be as big as possible in both axes. If either axis is
-/// unconstrained, then in that direction it will be sized to fit the child's
-/// dimensions. Using widthFactor and heightFactor you can force this latter
-/// behavior in all cases.
+/// This widget will be as big as possible if its dimensions are constrained and
+/// [widthFactor] and [heightFactor] are null. If a dimension is unconstrained
+/// and the corresponding size factor is null then the widget will match its
+/// child's size in that dimension. If a size factor is non-null then the
+/// corresponding dimension of this widget will be the product of the child's
+/// dimension and the size factor. For example if widthFactor is 0.5 then
+/// the width of this widget will always be half of the child's width.
 ///
 /// See also:
 ///
 ///  * [CustomSingleChildLayout]
 ///  * [Center] (which is the same as [Align] but with the [alignment] always
 ///    set to [FractionalOffset.center])
+///  * [FractionallySizedBox] which sizes its child based on its own size.
```

Comment:
we should be consistent about whether we put the description in parentheses or not (here and anywhere else we do this)